airflow/models.py

```diff
@@ -3032,6 +3032,25 @@ def get_task_instances(
     def roots(self):
         return [t for t in self.tasks if not t.downstream_list]
 
+    def topological_sort(self):
+        """
+        Sorts tasks in topographical order, such that a task comes after any of its
+        upstream dependencies.
+        """
+        stack, visited, sorted_tasks = [], set(), []
+
+        # begin with any tasks that have no upstream dependency
+        stack.extend(t for t in self.tasks if len(t.upstream_list) == 0)
+
+        while stack:
```

Comment:
I assume you say here A is dependent on both C and D? @jlowin as the "owner" of the algorithm would you mind taking a look?